scripts/delete_sample_crams.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+
+"""
+Given a bucket path and sample IDs, deletes cram files in
+the path for each sample listed.
+"""
+
+import logging
+import sys
+import subprocess
+import click
+
+
+def check_paths_exist(paths: list[str]):
+    """
+    Checks a list of gs:// paths to see if they point to an existing blob
+    Logs the invalid paths if any are found
+    """
+    invalid_paths = False
+    for path in paths:
+        # gsutil ls <path> returns '<path>\n' if path exists
+        result = subprocess.run(
+            ['gsutil', 'ls', path], check=True, capture_output=True, text=True
+        ).stdout.strip('\n')
+        if result == path:
+            continue
+        # If path does not exist, log the path and set invalid_paths to True
+        logging.info(f'Invalid path: {path}')
+        invalid_paths = True
+
+    if invalid_paths:
+        return False
+    return True
+
+
+def delete_from_bucket(paths: list[str]):
+    """
+    Delete CRAM and CRAM.crai files from bucket paths
+    """
+
+    subprocess.run(
+        ['gsutil', 'rm', *paths],
+        check=True,
+    )
+    logging.info(f'Deleted items: {paths}')
+
+
+@click.command()
+@click.option('--delete-path', '-d', help='GCP path to CRAMs to delete', required=True)
+@click.option(
+    '--somaliers',
+    '-s',
+    help='Also delete cram.somalier files',
+    is_flag=True,
+    default=False,
+)
+@click.argument('samples', nargs=-1)
+def main(delete_path: str, somaliers: bool, samples):
+    """
+
+    Parameters
+    ----------
+    delete_path :   a gcp path containing the crams to delete.
+                        e.g. gs://cpg-project-main/exome/cram
+    somaliers   :   flag if .cram.somalier files should be deleted too
+    samples     :   a list of sample ids to copy to the release bucket
+    """
+
+    sample_ids = list(samples)
+
+    # Generate the paths to the crams and crais to be deleted
+    cram_paths = []
+    for sample in sample_ids:
+        cram_paths.append(f'{delete_path}/{sample}.cram')
+        cram_paths.append(f'{delete_path}/{sample}.cram.crai')
+        if somaliers:
```

Comment:
--somaliers defaults to False, but I think this means we only get one shot at cleaning up the Somalier files at the same time:

- run for [samples]
- always add CRAM files
- [optionally add somalier files]
- all files need to exist to trigger the deletion

AFAIK this way we can never delete CRAMs first, and Somalier files separately.. should we default to clearing those up at the same time?

Reply:
Yeah that's a good point to consider. 
Are there any other cram file extensions that might exist? I'm noticing some cram.md5 files in one of the buckets. Maybe I should just change it so that anything that looks like `<sample_id>.cram.extension` is included in the scope of deletions